compiler_opt/rl/corpus.py

```diff
@@ -30,9 +32,47 @@ class ModuleSpec:
   """
   name: str
   exec_cmd: Tuple[str, ...] = ()
+  size: int = 0
 
 
-def build_modulespecs_from_datapath(
+class Corpus:
+  """Represents a corpus. Comes along with some utility functions."""
+
+  def __init__(self,
+               data_path: str,
+               additional_flags: Tuple[str, ...] = (),
+               delete_flags: Tuple[str, ...] = (),
+               module_specs: Optional[List[ModuleSpec]] = None):
+    if module_specs is not None:
+      self._module_specs = module_specs
+    else:
+      self._module_specs = _build_modulespecs_from_datapath(
+          data_path=data_path,
+          additional_flags=additional_flags,
+          delete_flags=delete_flags)
+    self.root_dir = data_path
+    self._module_specs.sort(key=lambda m: m.size, reverse=True)
+
+  def sample(self, k: int) -> List[ModuleSpec]:
+    """Samples `k` module_specs, sorting by size descending."""
+    k = min(len(self._module_specs), k)
+    if k < 1:
+      raise ValueError('Attempting to sample <1 module specs from corpus.')
+    return list(
```

Comment:
nit: can you write sampling with one line of code, and sorting with another line of code to emphasize that the sorting does not affect the sampling result?